rc/filetype/just.kak

```diff
@@ -39,6 +39,8 @@ define-command -hidden just-indent-on-new-line %{
 
 add-highlighter shared/justfile regions
 add-highlighter shared/justfile/content default-region group
+add-highlighter shared/justfile/recipe region '^[\w-]+(\s.+)?\s?:(?!=).*' '$' group
```

Comment:
Is the intent to match across lines? Maybe `\s` could be turned into `\h` to make it clearer.

And why declare a region, when this line could be a one-line regex? For clarity, or is there a technical obstacle?

Reply:
No the intent is only to match one single line (the recipe line, similar to a `make` target line). `\s` is indeed a single white space within the recipe regex.

> And why declare a region, when this line could be a one-line regex? For clarity, or is there a technical obstacle?

Further down, I declare an operator against this region. since there can be multiple arguments, with or without operators, the region regex would become overly complex: right now args are simly catched by: `(\s.+)?\s?`

and then:
`add-highlighter shared/justfile/recipe/ regex '([=+])' 1:operator` - simple but effective

Reply:
`ssh ctx="current": (_check ctx)` &mdash; example recipe line

Reply:
I'd use `[^\n]` instead of `.` to prevent matching newlines. Also, quotes are
usually not necessary for regex literals. I believe it can be simplified to
(untested)

```
add-highlighter shared/justfile/recipe region ^[\w-]+(\h[^\n]*|\h?):(?!=) $ group
```

I tink comments are not highlighted correctly in `ssh: # comment`.

Maybe you can do without a region, and just match all parts in one regex.

Reply:
Oh I was confused about `\s` & `\h` - indeed it should be `\h`